tests/unit/test_units.py

```diff
@@ -156,3 +156,11 @@ def test_when_tartget_unit_is_empty_string_raises_error(rule_with_units, from_un
     da = xr.DataArray(10, attrs={"units": from_unit})
     with pytest.raises(ValueError):
         handle_unit_conversion(da, rule_spec)
+
+
+def test_units_with_psu():
+    da = xr.DataArray(10, attrs={"units": "psu"})
+    new_da = da.pint.quantify("psu")
+    new_da = new_da.pint.to("0.001").pint.dequantify()
+    assert new_da.data == np.array(10)
+    assert new_da.attrs.get("units") == "0.001"
```

Comment:
if the intension is to rename units without changing the magnitude, then da.attrs['units'] = "0.001" is the way to go.

Reply:
No, the point is to demonstrate that this is what your code automatically tries to do in `units.py`, and leads to a crash. So, we need to have a way to handle this case. You could abstract this test into one that uses the actual unit conversion function, and then handles the conversion correctly.